src/couch/src/couch_httpd_auth.erl

```diff
@@ -662,10 +643,14 @@ authenticate(Pass, UserProps) ->
                     couch_util:get_value(<<"password_sha">>, UserProps, nil)
                 };
             <<"pbkdf2">> ->
+                PRF = couch_util:get_value(<<"pbkdf2_prf">>, UserProps, <<"sha">>),
+                verify_prf(PRF),
                 Iterations = couch_util:get_value(<<"iterations">>, UserProps, 10000),
```

Comment:
Since we bumped iterations to 50000, should this be updated as well?

Reply:
yes, not sure how I missed that. thanks

Reply:
hm, seconds thoughts on this. the 10000 setting is quite old and for the case where UserProps has pbkdf2 hash but has somehow lost its "iterations" property. Frankly having a default here makes no sense, so perhaps it should be a get_value/2 and then verify_iterations tweaked to fail for `undefined`?

The get_value/3 for pbkdf2_prf on the other hand _is_ right. existing pbkdf2 hashes will be hashed with `sha` and won't have this new field.

Reply:
That is an odd one... did we ever make a release which always assumed a hard-code iterations value of 10000, if there is a slight chance we did, we should probably keep the value as is. If you recall better and there is no chance we'd actually rely on that default 10000 then maybe doing the `undefined` or `nil` verification would make sense.